File: frontend/angular-ecommerce/src/app/common/order-item.ts
import { CartItem } from "./cart-item";

export class OrderItem {
    imageUrl!:string;
    unitPrice!:number;
    quantity!:number;
    productId!:string;

    //將訂單內容初始化，將購物車內的放入
    constructor(cartItem:CartItem){
        this.imageUrl = cartItem.imageUrl;
        this.unitPrice = cartItem.unitPrice;
        this.quantity = cartItem.quantity;
        this.productId = cartItem.id;
    }
}

File: backend/ecommerce/src/main/java/com/ecommerce/entity/OrderItem.java
package com.ecommerce.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Table(name = "order_item")
@Data
public class OrderItem {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "quantity")
    private int quantity;

    @Column(name = "unit_price")
    private BigDecimal unitPrice;


    @Column(name = "product_id")
    private Long productId;

    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;


}
